### PR TITLE
Implement Hybrid dependencies (#241)

### PR DESCRIPTION
**Pull request by inikulin**
\cc @AlexanderMoskovkin @helen-dikareva 
@VasilyStrelyaev please, check comments and strings

### CODE REVIEW

File: test/functional/site/index.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-    server1.close();
-    server2.close();
-};
+var Server = require('./server');
+
+
+var server1 = null;
+var server2 = null;
+
+exports.getCode = function (port1, port2, viewsPath) {
```
Comment (AlexanderMoskovkin) [inline test/functional/site/index.js]: it seems this is changed by a mistake

File: test/server/compiler-test.js [hunk 2]
```diff
@@ -666,6 +666,53 @@ describe('Compiler', function () {
...
+                        callsite: "   1 |import { Hybrid } from 'testcafe';\n" +
+                                  '   2 |\n' +
+                                  '   3 |fixture `Test`;\n' +
+                                  '   4 |\n' +
+                                  " > 5 |var selectYo = Hybrid(() => document.querySelector('#yo'), '42');\n"
+                    });
+                });
+        });
+
+        it('Should raise error if Hybrid function dependency is not a Hyybrid function', function () {
```
Comment (AlexanderMoskovkin) [inline test/server/compiler-test.js]: a typo in the `Hyybrid` word

File: test/functional/site/index.js (hunk 1 above)
Comment (inikulin) [inline test/functional/site/index.js]: Good catch! Auto-renaming...

File: src/api/common/hybrid/compiled-code-symbol.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// NOTE: This symbol used for storing hybrid function's compiled code for it's
+// later use as dependency. Also, it's used a hybrid function marker.
```
Comment (AlexanderMoskovkin) [inline src/api/common/hybrid/compiled-code-symbol.js]: as i understand it should be "Also, it's used **as** a hybrid function marker"

**Comment by inikulin** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm

File: src/compiler/es-next/compile-hybrid-function.js [hunk 4]
```diff
@@ -75,25 +76,38 @@ function downgradeES (fnCode) {
...
             return polyfillsCode;
         }, '');
 
-    return `(function(){${polyfills} return ${fnCode}})();`;
+    return { polyfills, modifiedFnCode };
 }
 
-export default function compileHybridFunction (fnCode) {
+function getDependenciesCode (dependencies) {
+    return Object.keys(dependencies).reduce((code, name) => {
```
Comment (helen-dikareva) [inline src/compiler/es-next/compile-hybrid-function.js]: I prefer style like in `polyfill` above
```
Object
.keys(dependencies)
.reduce((code, name) => {
```

**Comment by inikulin** [comment]
FPR

**Comment by helen-dikareva** [comment]
lgtm

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

File: src/api/common/hybrid/compiled-code-symbol.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+// NOTE: This symbol used for storing hybrid function's compiled code for it's
+// later use as dependency. Also, it's used as a hybrid function marker.
```
Comment (VasilyStrelyaev) [inline src/api/common/hybrid/compiled-code-symbol.js]:
```
// NOTE: This symbol is used for storing the hybrid function's compiled code for its
// later use as a dependency. Also, it's used as a hybrid function marker.
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by inikulin** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by inikulin** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Pull request merged**